flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java

```diff
@@ -630,66 +715,75 @@ public CommandLine getCommandLine(final Options commandOptions, final String[] a
 	}
 
 	/**
-	 * Executes the SAVEPOINT action.
-	 *
-	 * @param args Command line arguments for the savepoint action.
+	 * The 'savepoint' action.
 	 */
-	protected void savepoint(String[] args) throws Exception {
-		LOG.info("Running 'savepoint' command.");
+	protected class ActionSavepoint implements CommandAction<SavepointOptions> {
 
-		final Options commandOptions = CliFrontendParser.getSavepointCommandOptions();
+		@Override
+		public SavepointOptions parse(String[] args) throws CliArgsException {
+			final Options commandOptions = CliFrontendParser.getSavepointCommandOptions();
 
-		final Options commandLineOptions = CliFrontendParser.mergeOptions(commandOptions, customCommandLineOptions);
+			final Options commandLineOptions = CliFrontendParser.mergeOptions(commandOptions, customCommandLineOptions);
 
-		final CommandLine commandLine = CliFrontendParser.parse(commandLineOptions, args, false);
+			final CommandLine commandLine = CliFrontendParser.parse(commandLineOptions, args, false);
 
-		final SavepointOptions savepointOptions = new SavepointOptions(commandLine);
+			return new SavepointOptions(commandLine);
+		}
 
-		// evaluate help flag
-		if (savepointOptions.isPrintHelp()) {
-			CliFrontendParser.printHelpForSavepoint(customCommandLines);
-			return;
+		@Override
+		public Configuration getConfiguration(
+			SavepointOptions options,
+			CustomCommandLine activeCommandLine) throws Exception {
+			return getEffectiveConfiguration(activeCommandLine, options.getCommandLine());
 		}
 
-		final CustomCommandLine activeCommandLine = validateAndGetActiveCommandLine(commandLine);
+		@Override
+		public int runAction(
+			SavepointOptions options,
+			Configuration configuration) throws Exception {
+			LOG.info("Running 'savepoint' command.");
 
-		if (savepointOptions.isDispose()) {
-			runClusterAction(
-				activeCommandLine,
-				commandLine,
-				clusterClient -> disposeSavepoint(clusterClient, savepointOptions.getSavepointPath()));
-		} else {
-			String[] cleanedArgs = savepointOptions.getArgs();
+			if (options.isDispose()) {
+				runClusterAction(
+					configuration,
+					clusterClient -> disposeSavepoint(clusterClient, options.getSavepointPath()));
+			} else {
```

Comment:
@V1ncentzzZ the indention "problem" is caused by this `else` bock.

Reply:
what's the progress on this PR？

Reply:
@jiasheng55  @V1ncentzzZ  Any progress about this？

Reply:
Are you referring to the indention "problem" or the review progress of this PR? If the former, I think it's not a problem. If the latter, I'm waiting for reviews from Flink committers too...

Reply:
this PR ,looking forward to reviews from Flink committers @V1ncentzzZ